src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -3086,5 +3086,54 @@ private static bool DeclareSameIdentifiers(SyntaxToken[] oldVariables, SyntaxTok
         }
 
         #endregion
+
+        protected override bool IsRudeEditDueToPrimaryConstructor(ISymbol symbol, CancellationToken cancellationToken)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.NamedType:
+                    break;
+
+                case SymbolKind.Parameter:
+                    {
+                        var container = symbol.ContainingSymbol;
+
+                        if (container is IMethodSymbol { IsImplicitlyDeclared: false, MethodKind: MethodKind.Constructor })
+                        {
+                            foreach (var syntaxReference in container.DeclaringSyntaxReferences)
+                            {
+                                if (syntaxReference.GetSyntax(cancellationToken) is
+                                    ClassDeclarationSyntax { ParameterList: not null } or
+                                    StructDeclarationSyntax { ParameterList: not null })
+                                {
+                                    return true;
+                                }
+                            }
+                        }
+                    }
+                    break;
+
+                default:
+                    {
+                        var container = symbol.ContainingSymbol;
+
+                        if (container is { Kind: SymbolKind.NamedType, IsImplicitlyDeclared: false })
```

Comment:
>if [](http://example.com/codeflow?start=24&length=2)

Is this treating edits to members as rude edit when the containing type has a primary constructor?

If this is a temporary restriction, consider adding a `PROTOTYPE` comment.

Reply:
> Is this treating edits to members as rude edit when the containing type has a primary constructor?

Yes.

> If this is a temporary restriction, consider adding a PROTOTYPE comment.

The comments are added at the call sites. The whole method is temporary.